### PR TITLE
MIG-747: Cached clients (MTC 1.5.0)

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-747

New module for enabling cached clients.

4.5+

Preview: https://deploy-preview-34148--osdocs.netlify.app/openshift-enterprise/latest/migration-toolkit-for-containers/advanced-migration-options-mtc.html#migration-enabling-cached-kubernetes-clients_advanced-migration-options-mtc

Approved by QE.

### CODE REVIEW

File: modules/migration-enabling-cached-kubernetes-clients.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+= Enabling cached Kubernetes clients
+
+You can enable cached Kubernetes clients in the `MigrationController` custom resource (CR) for improved performance during migration. The greatest performance benefit is displayed when migrating between clusters in different regions or with significant network latency.
+
+[NOTE]
+====
+Delegated tasks, for example, Rsync backup for direct volume migration or Velero backup and restore, however, do not show improved performance with cached clients.
+====
+
+Cached clients require extra memory because the `MigrationController` CR caches all API resources required for interacting with `MigCluster` CRs. Requests that would normally go to the API server are directed to the cache instead. The cache watches the API server for updates.
```
Comment (apinnick) [inline modules/migration-enabling-cached-kubernetes-clients.adoc:16]: @eriknelson @djwhatle 
Can we be specific about how much extra memory is required to enable cached clients?

**Comment by XinRedhat** [comment]
@apinnick LGTM

File: modules/migration-enabling-cached-kubernetes-clients.adoc (hunk 1 above)
Comment (mikemckiernan) [inline modules/migration-enabling-cached-kubernetes-clients.adoc:16]: nit: Does "Requests that are normally sent to the API server..." read better?

nit 2: maybe "...caches all API resources that are required..."

File: modules/migration-enabling-cached-kubernetes-clients.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+You can increase the memory limits and requests of the `MigrationController` CR if `OOMKilled` errors occur after you enable cached clients.
+
+.Procedure
+
+. Enable cached clients by running the following command:
++
+[source,terminal]
+----
+$ oc --namespace openshift-migration patch MigrationController migration-controller \
```
Comment (mikemckiernan) [inline modules/migration-enabling-cached-kubernetes-clients.adoc:26]: Please lowercase `migrationcontroller` like the preceding topic on the HTML page: [Enabling persistent volume resizing,,,](https://deploy-preview-34148--osdocs.netlify.app/openshift-enterprise/latest/migration-toolkit-for-containers/advanced-migration-options-mtc.html#migration-enabling-pv-resizing-dvm_advanced-migration-options-mtc)

**Review by mikemckiernan — APPROVED** [review]
Some suggestions.  Please let me know if I can clarify my comments.

File: modules/migration-enabling-cached-kubernetes-clients.adoc (hunk 1 above)
Comment (apinnick) [reply modules/migration-enabling-cached-kubernetes-clients.adoc:16]: Yes, the cache watches the API server because that is the data that is cached.

**Pull request merged**